test-data/unit/check-typeddict.test

```diff
@@ -813,3 +813,29 @@ p = TaggedPoint(type='2d', x=42, y=1337)
 p.get('x', 1 + 'y')     # E: Unsupported operand types for + ("int" and "str")
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-full.pyi]
+
+
+-- Special cases
```

Comment:
Good comments! My responses:
* This seems to remove the only remaining join during semantic analysis. Updated fixed issues in PR description.
* Added  #3575 about using join with tuples. This seems to affect ordinary tuples as well, not just named tuples.
* I'll add a test case using the class syntax.